examples/petsc/src/petscutils.c

```diff
@@ -135,6 +136,7 @@ PetscErrorCode SetupDMByDegree(DM dm, PetscInt p_degree, PetscInt q_extra,
     ierr = DMAddBoundary(dm, DM_BC_ESSENTIAL, "wall", label, 1,
                          marker_ids, 0, 0, NULL, (void(*)(void))bc_func,
                          NULL, NULL, NULL); CHKERRQ(ierr);
+    ierr = DMViewFromOptions(dm, NULL, "-dm_view"); CHKERRQ(ierr);
```

Comment:
Only the first DMViewFromOptions will happen - I was fighting this last week trying to debug something in Ratel

```
  PetscCall(DMSetOptionsPrefix(*dm, "final_"));
  PetscCall(DMViewFromOptions(*dm, NULL, "-dm_view"));
```

Reply:
Interesting, then I can call with  `-final_dm_view`?

Reply:
exactly (Note that Ratel code has that extra deref which you don't need)

Reply:
Thanks, for `bp1` and `bp2` we don't call that block, I tried with `bp3` and we have 
```
 ./bps -problem bp3 -degree 3 -final_dm_view -simplex
DM Object: (final_) 1 MPI process
  type: plex
DM_0x84000001_3 in 3 dimensions:
  Number of 0-cells per rank: 64
  Number of 1-cells per rank: 279
  Number of 2-cells per rank: 378
  Number of 3-cells per rank: 162
Labels:
  marker: 1 strata with value/size (1 (326))
  celltype: 4 strata with value/size (0 (64), 6 (162), 3 (378), 1 (279))
  depth: 4 strata with value/size (0 (64), 1 (279), 2 (378), 3 (162))
Field P3:
  adjacency FEM
```

Reply:
And `ksp` diverges.

Reply:
So, `bp1` and `bp2` works fine and error reduces with refinement, we need to increase `-ksp_max_it`. 
But when we apply `bc` and do refinement the error doesn't get smaller

Reply:
Why error increases with `local_nodes 2000`? Weird!

500
```
./bps -problem bp5 -degree 3 -simplex -local_nodes 500 -ksp_max_it 1000

  KSP:
    KSP Type                                : cg
    KSP Convergence                         : CONVERGED_RTOL
    Total KSP Iterations                    : 46
    Final rnorm                             : 1.948723e-10
  Performance:
    Pointwise Error (max)                   : 1.207972e-01
    CG Solve Time                           : 0.00755058 (0.00755058) sec
    DoFs/Sec in CG                          : 1.92515 (1.92515) million
```
1000
```
./bps -problem bp5 -degree 3 -simplex -local_nodes 1000 -ksp_max_it 1000

  KSP:
    KSP Type                                : cg
    KSP Convergence                         : CONVERGED_RTOL
    Total KSP Iterations                    : 73
    Final rnorm                             : 1.748494e-10
  Performance:
    Pointwise Error (max)                   : 5.195360e-02
    CG Solve Time                           : 0.0309736 (0.0309736) sec
    DoFs/Sec in CG                          : 1.81477 (1.81477) million
```

2000
```
 ./bps -problem bp5 -degree 3 -simplex -local_nodes 2000 -ksp_max_it 1000
 
   KSP:
    KSP Type                                : cg
    KSP Convergence                         : CONVERGED_RTOL
    Total KSP Iterations                    : 80
    Final rnorm                             : 1.360241e-10
  Performance:
    Pointwise Error (max)                   : 3.102214e-01
    CG Solve Time                           : 0.0418808 (0.0418808) sec
    DoFs/Sec in CG                          : 2.9646 (2.9646) million
```

4000
```
./bps -problem bp5 -degree 3 -simplex -local_nodes 4000 -ksp_max_it 1000

  KSP:
    KSP Type                                : cg
    KSP Convergence                         : CONVERGED_RTOL
    Total KSP Iterations                    : 90
    Final rnorm                             : 8.223407e-11
  Performance:
    Pointwise Error (max)                   : 1.360253e-02
    CG Solve Time                           : 0.1037 (0.1037) sec
    DoFs/Sec in CG                          : 2.8918 (2.8918) million
```

Reply:
How are you calculating error in the max norm? Also, do you know the asymptotic convergence rate for that one?

Reply:
Here is the code for computing the error. `collocated_error` is: u_h-u_e
```
  // Reduce max error
  *max_error = 0;
  const CeedScalar *e;
  CeedVectorGetArrayRead(collocated_error, CEED_MEM_HOST, &e);
  for (CeedInt i=0; i<length; i++) {
    *max_error = PetscMax(*max_error, PetscAbsScalar(e[i]));
  }
  CeedVectorRestoreArrayRead(collocated_error, &e);
  ierr = MPI_Allreduce(MPI_IN_PLACE, max_error, 1, MPIU_REAL, MPIU_MAX,
                       user->comm);
  CHKERRQ(ierr);
```

Reply:
I don't think this is an accurate way to evaluate L_\infty estimates. There is a pretty good way in "Optimal L_\infty Estimates for the Finite Element Method on Irregular Meshes" by Ridgway.

Reply:
@rezgarshakeri, to confirm, from the discussion below I understand that you tried computing the true solution on the nodes instead of the quadrature points (either by computing the true solution in a collocated quadrature space and accounting for multiplicity and then taking the difference at the nodes or projecting the solution back onto the nodes like in Ratel and then taking the difference at the nodes) and it did not change the error you were seeing?

If that is the case, I think we should stop letting this PR hold back the release. I don't know where else the issue would be.

Reply:
Do we want to compute error in the nodes for hex as well or just tet?

Reply:
I think its easier to do it the same for both

Reply:
Could you please tell which part of ratel does the projection to the nodes?

Reply:
It does it as part of the diagnostic values output